app/src/main/java/ch/heigvd/pro/b04/android/Utils/Persistent.java

```diff
@@ -0,0 +1,44 @@
+package ch.heigvd.pro.b04.android.Utils;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import ch.heigvd.pro.b04.android.R;
+import ch.heigvd.pro.b04.android.Utils.Exceptions.TokenNotSetException;
+
+public final class Persistent {
+    private Persistent() {};
+
+    /**
+     * This function returns the token if it was set previously.
+     * @param context The app context. Needed to access the Shared Preferences
+     * @return The token value
+     * @throws TokenNotSetException is thrown if no token was stored
+     */
+    public static String getStoredTokenOrError(Context context) throws TokenNotSetException {
+        String defValue = "Error";
+        SharedPreferences preferences = context.getSharedPreferences(
+                context.getString(R.string.data), Context.MODE_PRIVATE);
+        String token = preferences.getString(context.getString(R.string.dataToken), defValue);
```

Comment:
You probably want to use a normal String as the key, as resources can change based on configuration (for instance if a different locale is applied).

Reply:
AFAIK, the [android documentation](https://developer.android.com/training/data-storage/shared-preferences]) suggests using `R.string.xxx`. My guess is that it's easier to change, refactor and access for other devs, but I have nothing against using normal strings.
Let me know

Reply:
I will do that